### PR TITLE
Add tier preference to security index settings allowlist and update default tier preference

### PR DESCRIPTION
**Pull request by AthenaEryma**
This commit allows tier preference for the security system indices to be set using the Security Settings API, and adds validation to prevent using the `data_frozen` tier for security system indices.

Also updates the default tier preference to `data_hot,data_content`.

### CODE REVIEW

File: x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java [hunk 1]
```diff
@@ -72,6 +72,27 @@ public void testBasicWorkflow() throws IOException {
...
+                    "index.routing.allocation.include._tier_preference": "data_hot"
+                },
+                "security-profile": {
+                    "index.routing.allocation.include._tier_preference": "data_hot"
+                }
+            }
+            """);
+        Response resp = adminClient().performRequest(req);
+        assertOK(resp);
+        Request getRequest = new Request("GET", "/_security/settings");
```
Comment (jakelandis) [inline x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java:89]: should this return the default values for all 3 not explicitly set ?

**Review by jakelandis — COMMENTED** [review]
looking good a couple questions.

File: x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java [hunk 2]
```diff
@@ -72,6 +72,27 @@ public void testBasicWorkflow() throws IOException {
...
+        Response getResp = adminClient().performRequest(getRequest);
+        assertOK(getResp);
+        final XContentTestUtils.JsonMapView mapView = createJsonMapView(getResp.getEntity().getContent());
+        assertThat(mapView.get("security.index.routing.allocation.include._tier_preference"), equalTo("data_hot"));
```
[the remark this answers is not in the document]
Comment (gwbrown) [reply x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java:93]: Ah yep, I copy/pasted this from the test above, I'll look to fix that one too.

File: x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java (hunk 1 above)
Comment (gwbrown) [reply x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java:90]: No, it's basically a thin proxy over the regular settings API.

File: x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java [hunk 3]
```diff
@@ -70,6 +70,30 @@ public void testBasicWorkflow() throws IOException {
...
+        assertThat(mapView.get("security-profile.index.routing.allocation.include._tier_preference"), equalTo("data_hot"));
+
```
Comment (jakelandis) [inline x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/SecuritySettingsIT.java:96]: nit: should we test setting this to `null` to restore the default ?

**Review by jakelandis — APPROVED** [review]
LGTM (one testing nitpick/question)

(sorry for the delay and thanks for the ping)

**Pull request merged**